apm-agent-plugins/apm-logging-plugin/apm-jul-plugin/src/main/java/co/elastic/apm/agent/jul/reformatting/JulEcsReformattingHelper.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package co.elastic.apm.agent.jul.reformatting;
+
+import co.elastic.apm.agent.loginstr.reformatting.AbstractEcsReformattingHelper;
+import co.elastic.apm.agent.loginstr.reformatting.Utils;
+import co.elastic.apm.agent.sdk.logging.Logger;
+import co.elastic.apm.agent.sdk.logging.LoggerFactory;
+import co.elastic.apm.agent.util.LoggerUtils;
+import co.elastic.logging.AdditionalField;
+import co.elastic.logging.jul.EcsFormatter;
+
+import javax.annotation.Nullable;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.logging.ConsoleHandler;
+import java.util.logging.FileHandler;
+import java.util.logging.Formatter;
+import java.util.logging.StreamHandler;
+
+class JulEcsReformattingHelper extends AbstractEcsReformattingHelper<StreamHandler, Formatter> {
+
+    private static final Logger logger = LoggerFactory.getLogger(JulEcsReformattingHelper.class);
+    private static final Logger oneTimeLogFileLimitWarningLogger = LoggerUtils.logOnce(logger);
+
+    private static final ThreadLocal<String> currentPattern = new ThreadLocal<>();
+    private static final ThreadLocal<Path> currentExampleLogFile = new ThreadLocal<>();
+
+    JulEcsReformattingHelper() {}
+
+    public boolean onAppendEnter(FileHandler fileHandler, String pattern, File exampleLogFile) {
+        try {
+            currentPattern.set(pattern);
+            currentExampleLogFile.set(exampleLogFile.toPath());
```

Comment:
If you look at `java.util.logging.FileHandler#openFiles`, you'll see that all files are created based on a single pattern. We only use this for the enclosing log directory. If the user makes the pattern so that directories get changed between logs, that is unexpected and we will just use the first dir. I don't think we need to add special handling for such case, but we can improve based on feedback.